src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -89,6 +91,18 @@ public BoundLambda(SyntaxNode syntax, UnboundLambda unboundLambda, BoundBlock bo
             );
         }
 
+        internal BoundLambda WithInAnonymousFunctionConversion()
+        {
+            if (InAnonymousFunctionConversion)
+            {
+                return this;
+            }
+
+            var result = (BoundLambda)MemberwiseClone();
```

Comment:
I expected we would just be able to set 'WasConverted', only in debut mode, and move on, rather than introducing an actual new field here and cloning the BoundLambda. Did I misunderstand something?

Reply:
It seems there was a misunderstanding, then. I think we cannot use `WasConverted` in Debug mode - we need to know whether the lambda was converted or not at runtime (even in Relase mode) to decide whether we should visit it from the `public VisitLambda` in the errorneous without-conversion case (otherwise we would visit every normal lambda twice). Put another way: The new runtime `BoundNode` flag replaces the private field `NullableWalker._visitingConversionOperand` added previously in this PR. We need to store that information somewhere if we want to go with this PR - as I don't think the `BoundLambda` today can tell us it's been converted or not.

Reply:
Of course. Makes complete sense. Thanks!